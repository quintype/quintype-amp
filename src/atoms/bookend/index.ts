import { Bookend } from "./bookend";
export { Bookend };
