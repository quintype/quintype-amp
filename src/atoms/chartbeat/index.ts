import { ChartBeat } from "./chartbeat";

export { ChartBeat };
