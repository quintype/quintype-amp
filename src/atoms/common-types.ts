import { CarouselTypes } from "./carousel/types";
import { FacebookTypes } from "./facebook/types";
import { TwitterTypes } from "./twitter/types";
import { InstagramTypes } from "./instagram/types";

declare global {
  namespace JSX {
    interface IntrinsicElements {
      "amp-img": any;
      "amp-carousel": CarouselTypes;
      "amp-facebook": FacebookTypes;
      "amp-twitter": TwitterTypes;
      "amp-sidebar";
      "amp-social-share";
      "amp-youtube";
      "amp-ad";
      "amp-iframe";
      "amp-dailymotion";
      "amp-mathml";
      "amp-instagram": InstagramTypes;
      "amp-web-push";
      "amp-web-push-widget";
      "amp-o2-player";
      "amp-analytics";
      "amp-nested-menu";
      "amp-next-page";
      "amp-live-list";
      "amp-story-grid-layer";
      "amp-story-page";
      "amp-story";
      "amp-video";
      "amp-story-bookend";
      "amp-story-auto-ads"
    }
  }
}

// Should only contain the common attributes given in https://amp.dev/documentation/guides-and-tutorials/learn/common_attributes/?format=websites
// do not add anything else here
export interface Common {
  fallback?: string;
  heights?: string;
  layout?: "nodisplay" | "fixed" | "responsive" | "fixed-height" | "fill" | "container" | "flex-item" | "intrinsic";
  media?: string;
  noloading?: boolean;
  on?: string;
  placeholder?: string;
  sizes?: string;
  width?: string;
  height?: string;
}
