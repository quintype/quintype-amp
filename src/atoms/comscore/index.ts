import { ComScore } from "./comscore";
export { ComScore };
