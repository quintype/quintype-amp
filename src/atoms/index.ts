import Carousel from "./carousel";
import { Image } from "./image";
import Layout from "./layout";
import Facebook from "./facebook";
import Twitter from "./twitter";
import { Footer } from "./footer";
import { Section } from "./section";
import { Spacer } from "./spacer";
import { Author } from "./author";
import { DateTime } from "./date-time";
import { StoryElement } from "./story-elements/story-element";
import { StoryElements } from "./story-elements";

export {
  Carousel,
  Image,
  Layout,
  Facebook,
  Twitter,
  Footer,
  Section,
  Spacer,
  Author,
  DateTime,
  StoryElement,
  StoryElements
};
