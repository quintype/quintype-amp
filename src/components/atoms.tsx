import React, { Fragment } from 'react'
import { Helmet } from 'react-helmet'
import * as atomTypes from '../type-declarations/atomicComponents'

export const Image = (props: atomTypes.Image) => {
  return <amp-img {...props}></amp-img>
}

export const Title = (props: atomTypes.Title) => {
  return (
    <Helmet>
      <title>{props.children}</title>
    </Helmet>
  )
}

export const Carousel = (props: atomTypes.Carousel) => {
  return (
    <Fragment>
      <Helmet>
        <script async={undefined} custom-element="amp-carousel" src="https://cdn.ampproject.org/v0/amp-carousel-0.2.js"></script>
        <meta name="lorem ipsum" content="carousel script added" />
      </Helmet>
      <amp-carousel {...props}>{props.children}</amp-carousel>
    </Fragment>
  )
}

export const FitText = (props: atomTypes.FitText) => {
  return (
    <Fragment>
      <Helmet>
        <script async={undefined} custom-element="amp-fit-text" src="https://cdn.ampproject.org/v0/amp-fit-text-0.1.js"></script>
        <meta name="lorem ipsum" content="Fit Text script added" />
      </Helmet>
      <amp-fit-text {...props}>{props.children}</amp-fit-text>
    </Fragment>
  )
}

export const Youtube = (props: atomTypes.Youtube) => {
  return (
    <Fragment>
      <Helmet>
        <script async={undefined} custom-element="amp-youtube" src="https://cdn.ampproject.org/v0/amp-youtube-0.1.js"></script>
        <meta name="lorem ipsum" content="YOUTUBE script added" />
      </Helmet>
      <amp-youtube {...props}>{props.children}</amp-youtube>
    </Fragment>
  )
}

// const Amp = {Image, Title, Carousel, FitText, Youtube}
// export default Amp