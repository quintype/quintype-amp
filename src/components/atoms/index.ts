import Carousel from './Carousel'
import FitText from './FitText'
import Image from './Image'
import Title from './Title'
import Youtube from './Youtube'

export {Carousel,FitText,Image,Title,Youtube}
