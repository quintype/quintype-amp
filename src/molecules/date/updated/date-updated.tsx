import React from "react";
import { DateTime } from "../../../atoms";
import { withStoryAndConfig } from "../../../context";
import get from "lodash.get";
import formatDistanceToNow from "date-fns/formatDistanceToNow";
import { utcToZonedTime } from "date-fns-tz";

const DateUpdatedBase = ({ story, config }) => {
  const showUpdateDate = get(config, ["opts", "headerCardConfig", "dateConfig", "showUpdateDate"], false);
  const updateDatePrepend = get(config, ["opts", "headerCardConfig", "dateConfig", "updateDatePrepend"], null);
  const updatedAt = story["updated-at"];
  if (!showUpdateDate || !updatedAt) return null;

  const timeZonedTime = utcToZonedTime(updatedAt, "Asia/Kolkata");
  const humanizedString = formatDistanceToNow(timeZonedTime, { addSuffix: true });

  return <DateTime formattedDate={humanizedString} prepend={updateDatePrepend} />;
};

export const DateUpdated = withStoryAndConfig(DateUpdatedBase);
