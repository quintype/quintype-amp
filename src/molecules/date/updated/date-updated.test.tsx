import React from "react";
import { DateUpdatedBase } from "./date-updated";
import { shallow } from "enzyme";
import { utcToZonedTime } from "date-fns-tz";
import formatDistanceToNow from "date-fns/formatDistanceToNow";
import { DateTime } from "../../../atoms";

describe("DateUpdated", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(<DateUpdatedBase story={getDummyStory()} prepend={"Story Updated:"} />);
    expect(wrapper).toMatchSnapshot();
  });
  it("should pass correct date to <DateTime />", () => {
    const story = getDummyStory();
    const wrapper = shallow(<DateUpdatedBase story={story} />);
    const timeZonedTime = utcToZonedTime(story["last-published-at"], "Asia/Kolkata");
    const humanizedString = formatDistanceToNow(timeZonedTime, { addSuffix: true });
    expect(wrapper.find(DateTime).prop("formattedDate")).toBe(humanizedString);
  });
  it("should pass prepend to <DateTime />", () => {
    const wrapper = shallow(<DateUpdatedBase story={getDummyStory()} prepend={"Story Updated:"} />);
    expect(wrapper.find(DateTime).prop("prepend")).toBe("Story Updated:");
  });
});

const getDummyStory = () => {
  return {
    "updated-at": 1581503848382,
    seo: {
      "meta-keywords": ["LSAC", "Law School", "legal Education", "Lawyers", "Advocates", "India", "Law"],
      "meta-title": "Empowering India’s Future Lawyers and Leaders",
      "meta-description":
        "#Sponsored: New milestones and initiatives mark LSAC’s ongoing efforts to increase access to legal education in India. - Bar & Bench ",
      "claim-reviews": {
        story: null
      }
    },
    "assignee-id": 708282,
    "author-name": "Ravi Kanth",
    tags: [
      {
        id: 2038546,
        name: "LSAC",
        "meta-description": null,
        "meta-title": null,
        slug: "lsac",
        "tag-type": "Tag"
      }
    ],
    headline: "#Sponsored: Empowering India’s Future Lawyers and Leaders",
    "storyline-id": null,
    votes: {},
    "story-content-id": "7f3d5bdb-ec52-4047-ac0d-df4036ec974b",
    slug: "news/empowering-indias-future-lawyers-and-leaders",
    "linked-stories": {},
    "last-published-at": 1590539519368,
    subheadline:
      "New milestones and initiatives mark LSAC’s ongoing efforts to increase access to legal education in India",
    alternative: {},
    sections: [
      {
        "domain-slug": null,
        slug: "news",
        name: "News",
        "section-url": "https://www.barandbench.com/news",
        id: 14017,
        "parent-id": null,
        "display-name": "News",
        collection: {
          slug: "news",
          name: "News",
          id: 29911
        },
        data: null
      }
    ],
    "read-time": 5,
    "access-level-value": null,
    "content-created-at": 1581326633010,
    "owner-name": "Ravi Kanth",
    "custom-slug": "Empowering India’s Future Lawyers and Leaders",
    "push-notification": null,
    "publisher-id": 829,
    "hero-image-metadata": {
      width: 2550,
      height: 1162,
      "mime-type": "image/jpeg",
      "file-size": 185634,
      "file-name": "Discover_Law_Logo_1.jpg",
      "focus-point": [1275, 581]
    },
    comments: null,
    "word-count": 1143,
    entities: {},
    "published-at": 1581503851412,
    "is-live-blog": false,
    "breaking-news-linked-story-id": null,
    "storyline-title": null,
    summary: null,
    "push-notification-title": null,
    "external-id": null,
    "canonical-url": null,
    "is-amp-supported": true,
    autotags: [],
    "linked-entities": [],
    status: "published",
    "hero-image-attribution": "Create",
    "bullet-type": "123",
    "hero-image-alt-text": null,
    id: "7f3d5bdb-ec52-4047-ac0d-df4036ec974b",
    "hero-image-s3-key": "barandbench/2020-02/43fb44c4-2028-4b90-95ad-dc54aad47a28/Discover_Law_Logo_1.jpg",
    contributors: null,
    "associated-series-collection-ids": [],
    cards: [],
    url: "https://www.barandbench.com/news/empowering-indias-future-lawyers-and-leaders",
    "story-version-id": "1110cdc2-a1d6-4a6c-9707-f4b9b516d49e",
    "content-type": "story",
    "content-updated-at": 1581503851699,
    "author-id": 708282,
    "owner-id": 708282,
    "linked-story-ids": ["a7a06a3f-8d9a-450c-8eaa-387b7c8c6323"],
    access: "subscription",
    "asana-project-id": null,
    "first-published-at": 1581503851412,
    "hero-image-caption": null,
    version: 12,
    "story-template": "text",
    "sequence-no": null,
    "created-at": 1581327522143,
    authors: [
      {
        id: 708282,
        name: "Ravi Kanth",
        slug: "ravi-kanth",
        "avatar-url":
          "https://images.assettype.com/barandbench/2020-01/82d5c350-9791-4cdb-88d5-f52b2d98a4ab/Bar_and_Bench_Icon_Purple.png",
        "avatar-s3-key": "barandbench/2020-01/82d5c350-9791-4cdb-88d5-f52b2d98a4ab/Bar_and_Bench_Icon_Purple.png",
        "twitter-handle": null,
        bio: null,
        "contributor-role": null
      }
    ],
    metadata: {
      "sponsored-by": "Empowering India’s Future Lawyers and Leaders",
      "card-share": {
        shareable: false
      }
    },
    "publish-at": null,
    "assignee-name": "Ravi Kanth"
  };
};
