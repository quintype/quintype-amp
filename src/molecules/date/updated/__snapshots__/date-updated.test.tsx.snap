// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateUpdated should match snapshot 1`] = `
<Component
  formattedDate="8 months ago"
  prepend="Story Updated:"
/>
`;
